wp-graphql.php

```diff
@@ -684,6 +690,45 @@ public static function get_app_context() {
 
 			return $app_context;
 		}
+
+		/**
+		 * Adds default GraphQL post-types settings
+		 *
+		 * @param WP_Post_Type|array $args       Post type arguments.
+		 * @param string             $post_type  Post type
+		 * 
+		 * @return array
+		 */
+		public static function add_post_type_settings( $args, $post_type ) {
+			global $wp_post_types;
+
+			$post_type_settings = [
+				'show_in_graphql'                 => true,
+				'graphql_single_name'             => str_replace( '_', '', ucwords( strtolower( $post_type ), '_' ) ),
+				'graphql_plural_name'             => str_replace( '_', '', ucwords( strtolower( $post_type ), '_' ) ) . 's',
+				'register_schema_type'            => true,
+				'register_root_query'             => true,
+				'register_root_connection'        => true,
+				'connect_to_comments'             => true,
+				'connect_to_taxonomies'           => true,
+				'supports_content_node'           => true,
+				'supports_post_object_union'      => true,
```

Comment:
Hmm. If you use `register_post_type()` and `show_in_graphql` your post type is already agreeing to this. 🤔

Reply:
Not necessarily, I'm forced to inject the `PostObjectUnion` using the `graphql_union_resolve_type` filter just to make it work with `Product` child types, if this is toggled to true.